vizro-core/src/vizro/managers/_data_manager.py

```diff
@@ -87,47 +63,39 @@ class _DynamicData:
 
     Possibly in future, this will become a public class so you could directly do:
         >>> data_manager["dynamic_data"] = DynamicData(dynamic_data, timeout=5)
-    But we'd need to make sure that name is not an argument in __init__ then.
 
     At this point we might like to disable the behavior so that data_manager setitem and getitem handle the same
     object rather than doing an implicit conversion to _DynamicData.
     """
 
-    def __init__(self, name: str, load_data: pd_DataFrameCallable):
+    def __init__(self, load_data: pd_DataFrameCallable):
         self.__load_data: pd_DataFrameCallable = load_data
-        # name is needed for the cache key and should not be modified by users.
-        self._name = name
         self.timeout: Optional[int] = None
         # We might also want a self.cache_arguments dictionary in future that allows user to customize more than just
         # timeout, but no rush to do this since other arguments are unlikely to be useful.
 
-    @memoize
-    def load(self) -> pd.DataFrame:
-        """Loads data.
-
-        In future this will probably take arguments that are passed through to __load_data in order to re-run the
-        loading function with different arguments. We might want to use CapturedCallable for self.__load_data then.
-        """
-        # Note you get the same "cache missed" message if NullCache is running or if data_manager._cache_has_app is
-        # False.
-        logger.debug("Cache miss; reloading data")
-        return self.__load_data()
-
-    def __repr__(self):
-        """Flask-caching uses repr to form the cache key, so this is very important to set correctly.
-
-        In particular, it must depend on something that uniquely labels the data source and is the same across all
-        workers: self._name. Using id(self), as in Python's default repr, only works in the case that gunicorn is
-        running with --preload: without preloading, the id of the same data source in different processes will be
-        different so the cache will not match up.
-
-        flask_caching make it possible to set a __cached_id__ attribute to handle this so that repr can be set
-        independently of cache key, but this doesn't seem to be well documented or work well, so it's better to rely
-        on __repr__.
-        """
-        # Note that using repr(self.__load_data) is not good since it depends on the id of self.__load_data and so
-        # would not be consistent across processes.
-        return f"{self.__class__.__name__}({self._name!r}, {self.__load_data.__qualname__})"
+    def load(self, *args, **kwargs) -> pd.DataFrame:
```

Comment:
This line is basically the key change; the other stuff is just needed to get flask-caching working properly.